packages/vue-client/src/views/GameView.vue

```diff
@@ -155,25 +147,41 @@ function makeMove(move_str: string) {
   requests.post(`/games/${props.gameId}/move`, move).catch(alert);
 }
 
+// subscribe to seat changes
+watchEffect((onCleanup) => {
+  const seatsMessage = `game/${props.gameId}/seats`;
+  socket.on(seatsMessage, (data) => {
+    players.value = data;
+  });
+  onCleanup(() => {
+    socket.off(seatsMessage);
+  });
+});
+
+// subscribe to game updates for specific seat
 watchEffect((onCleanup) => {
   if (!props.gameId) {
     return;
   }
-  const topic = `game/${props.gameId}`;
+  const topic =
+    playing_as.value == null
+      ? `game/${props.gameId}`
+      : `game/${props.gameId}/${playing_as.value}`;
   socket.emit("subscribe", [topic]);
-  socket.on("move", (state: GameStateResponse) => {
-    setNewState(state);
+
+  onCleanup(() => {
+    socket.emit("unsubscribe", [topic]);
```

Comment:
When a player selects a seat, does this make it so the previous topic gets unsubscribed?

Reply:
Yep!